tests/test-dirs/syntax-document/language-extensions.t/run.t

```diff
@@ -0,0 +1,247 @@
+  $ syn_doc 11:37 -filename ./sig-subs.ml < ./sig-subs.ml | jq '.value'
+  "No documentation found"
+On type t
+  $ syn_doc 2:9 -filename ./sig-subs.ml < ./sig-subs.ml | jq '.value.name'
+  "Abstract type"
+// Local substitutions
+  $ syn_doc 16:12 -filename ./sig-subs.ml < ./sig-subs.ml | jq '.value.name'
+  "Local substitution"
+// Module type substitutions
+  $ syn_doc 25:57 \
+  > -filename ./sig-subs.ml < ./sig-subs.ml | jq '.value.name'
+  "Module substitution"
+
+
+// Types
+  $ cat > types.ml << EOF 
+  > type a1 = ..
+  > type a2 = A
+  > type a3 = |
+  > type a4 = {x: int}
+  > type a5 = int
+  > EOF
+on type a1..
+  $ syn_doc 1:5 \
+  > -filename ./types.ml < ./types.ml | jq '.value.name'
+  "Extensible variant type"
+on type a2..
+  $ syn_doc 2:5 \
+  > -filename ./types.ml < ./types.ml | jq '.value.name'
+  "Variant Type"
+on type a3..
+  $ syn_doc 3:6 \
+  > -filename ./types.ml < ./types.ml | jq '.value.name'
+  "Empty Variant type"
+on type a4..
+  $ syn_doc 4:5 \
+  > -filename ./types.ml < ./types.ml | jq '.value.name'
+  "Record type"
+on type a5..
+  $ syn_doc 5:5 \
+  > -filename ./types.ml < ./types.ml | jq '.value'
+  "No documentation found"
+
+// Private types
+// Extensible
+  $ cat > p-types.ml << EOF
+  > type b1 = private ..
+  > type b2 = private A
+  > type b3 = private A of int
+  > type b4 = private { x:int }
+  > type b5 = private int 
+  > module N : sig
+  >   type t = private int
+  >   val of_int: int -> t
+  >   val to_int: t -> int
+  > end = struct
+  >   type t = int
+  >   let of_int n = assert (n >= 0); n
+  >   let to_int n = n
+  > end
+  > EOF
+  on type b1..
+  $ syn_doc 1:14 \
+  > -filename ./p-types.ml < ./p-types.ml | jq '.value.name'
+  "Private Extensible Variant Type"
+on type b2..
+  $ syn_doc 2:14 \
+  > -filename ./private-types.ml < ./p-types.ml | jq '.value.name'
+  "Private Type"
+on type b3..
+  $ syn_doc 3:14 \
+  > -filename ./p-types.ml < ./p-types.ml | jq '.value.name'
+  "Private Type"
+on type b4..
+  $ syn_doc 4:14 \
+  > -filename ./p-types.ml < ./p-types.ml | jq '.value.name'
+  "Private Record Type"
+on b5..
+  $ syn_doc 5:14 \
+  > -filename ./p-types.ml < ./p-types.ml | jq '.value.name'
+  "Private Type Abbreviation"
+on type t = private int..
+  $ syn_doc 7:14 \
+  > -filename ./p-types.ml < ./p-types.ml | jq '.value.name'
+  "Private Type Abbreviation"
+on type t = int..
+  $ syn_doc 11:7 \
+  > -filename ./p-types.ml < ./p-types.ml | jq '.value'
+  "No documentation found"
+
+
+// Locally abstract data types
+  $ cat > locally-abstract-dt.ml << EOF 
+  > let f = fun (type t) (x: t) -> x = x
+  > let sort_uniq (type s) (cmp : s -> s -> int) =
+  >   let module S = Set.Make(struct type t = s let compare = cmp end) in
+  >   fun l ->
+  >     S.elements (List.fold_right S.add l S.empty)
+  > EOF
+// Locally abstract data types
+on type t..
+  $ syn_doc 1:17 \
+  > -filename ./locally-abstract-dt.ml < ./locally-abstract-dt.ml | jq '.value.name'
+  "Locally Abstract Type"
```

Comment:
Note that it also triggers when the cursor is on the body of the function (`1:39`) which is not very satisfying.

We should at least show it with a test and add a FIXME comment before. 

We probably won't be able to improve on it without having additional location information from the compiler (it might even be already erased in the parsetree, we need a CST..)